src/app/qgshandlebadlayers.cpp

```diff
@@ -117,8 +123,8 @@ QgsHandleBadLayers::QgsHandleBadLayers( const QList<QDomNode> &layers )
     QString provider = node.namedItem( QStringLiteral( "provider" ) ).toElement().text();
     QString vectorProvider = type == QLatin1String( "vector" ) ? provider : tr( "none" );
     bool providerFileBased = ( QgsProviderRegistry::instance()->providerCapabilities( provider ) & QgsDataProvider::File ) != 0;
-    const QString basepath = datasource.left( datasource.lastIndexOf( '/' ) );
-    mFileBase[name].append( basepath );
+
+    mOriginalFileBase[name] = datasource.left( datasource.lastIndexOf( '/' ) );
```

Comment:
I wonder if the layer id would be more suitable than the layer name for this registry?

Reply:
If they are maintained for each layer and robust, yes.